test/integration/integration_stream_decoder.cc

```diff
@@ -107,7 +130,7 @@ void IntegrationStreamDecoder::decodeData(Buffer::Instance& data, bool end_strea
   saw_end_stream_ = end_stream;
   body_ += data.toString();
 
-  if (end_stream && waiting_for_end_stream_) {
+  if (end_stream && (waiting_for_reset_ || waiting_for_end_stream_)) {
     dispatcher_.exit();
   } else if (body_data_waiting_length_ > 0) {
     body_data_waiting_length_ -= std::min(body_data_waiting_length_, data.length());
```

Comment:
I'm a bit confused why we subtract the min of these two vs just data.length() here

Reply:
I guess because you can say "wait for 10 bytes" and you're still done waiting if 20 bytes came in.